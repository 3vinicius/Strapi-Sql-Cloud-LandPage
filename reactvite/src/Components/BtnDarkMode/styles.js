import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    height: 80px;
    width: 50px;
    background-color: red;
    padding: 10px;
    border-radius: 25px;
    margin: 10px;
    cursor: pointer;
  `}
`;
