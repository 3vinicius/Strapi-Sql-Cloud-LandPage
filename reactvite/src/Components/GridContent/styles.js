import styled, { css } from 'styled-components';
import darkMode from '../../Utils/stylesUtils';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
    text-align: center;
    max-width: 58rem;
    margin: auto;
    font-size: ${theme.fonts.sizes.medium};
    ${darkMode(colorDark)}
    ;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge}
  `}
`;
