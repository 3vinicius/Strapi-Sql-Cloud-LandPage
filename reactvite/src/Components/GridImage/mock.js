export default {
  background: false,
  title: 'My grid',
  description:
    'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Molestiae quisquam ex modi reprehenderit officia nam blanditiis animi laboriosam aperiam voluptatibus, fuga placeat architecto, dicta impedit non dolores repellat natus inventore!',
  grid: [
    {
      id: 1,
      title: 'accusamus beatae ad facilis cum similique qui sunt',
      url: 'https://via.placeholder.com/600/92c952',
    },
    {
      id: 2,
      title: 'reprehenderit est deserunt velit ipsam',
      url: 'https://via.placeholder.com/600/771796',
    },
    {
      id: 3,
      title: 'officia porro iure quia iusto qui ipsa ut modi',
      url: 'https://via.placeholder.com/600/24f355',
    },
    {
      id: 4,
      title: 'culpa odio esse rerum omnis laboriosam voluptate repudiandae',
      url: 'https://via.placeholder.com/600/d32776',
    },
    {
      id: 5,
      title: 'natus nisi omnis corporis facere molestiae rerum in',
      url: 'https://via.placeholder.com/600/f66b97',
    },
    {
      id: 6,
      title: 'accusamus ea aliquid et amet sequi nemo',
      url: 'https://via.placeholder.com/600/56a8c2',
    },
    {
      id: 7,
      title: 'officia delectus consequatur vero aut veniam explicabo molestias',
      url: 'https://via.placeholder.com/600/b0f7cc',
    },
    {
      id: 8,
      title: 'aut porro officiis laborum odit ea laudantium corporis',
      url: 'https://via.placeholder.com/600/54176f',
    },
    {
      id: 9,
      title: 'qui eius qui autem sed',
      url: 'https://via.placeholder.com/600/51aa97',
    },
    {
      id: 10,
      title: 'beatae et provident et ut vel',
      url: 'https://via.placeholder.com/600/810b14',
    },
  ],
};
