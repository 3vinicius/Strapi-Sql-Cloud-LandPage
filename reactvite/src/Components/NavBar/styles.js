import styled, { css } from 'styled-components';

export const Container = styled.ul`
  ${({ theme }) => css`
    background-color: white;
    display: flex;
    flex-direction: row;
    flex-wrap:wrap;
    margin: 0px;
    @media ${theme.myMedias.letMedium} {
      flex-flow: column wrap;
      justify-content: center;
      align-items: center;
    }
  `}
`;
