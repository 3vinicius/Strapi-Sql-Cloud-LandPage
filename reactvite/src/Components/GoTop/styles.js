import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.a`
  ${() => css``}
  position: fixed;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.colorWhite};
  display:flex;
  align-items: center;
  justify-content: center;
  height:4rem;
  width: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  transition: all 0.5s ease-out  ;
  border: 1px solid #060a18;
  box-shadow: 0px 0px 5px 2px #00000054;


  &:hover{
    opacity: 1;
  }
`;
