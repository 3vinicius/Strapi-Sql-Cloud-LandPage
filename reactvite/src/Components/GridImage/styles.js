import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
  padding: 200px
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    ;
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  overflow: hidden;
  `}
`;

export const Imagem = styled.img`
  ${() => css`
    width:100%;
    height: 100%;
    transition: all 300ms ease-in-out;

    &:hover{
      transform: scale(1.2)
    }
  `}
`;
