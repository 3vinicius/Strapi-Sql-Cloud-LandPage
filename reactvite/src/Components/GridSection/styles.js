import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme, colorDark }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacings.xhuge};
    background: ${
      colorDark ? theme.colors.primaryColor : theme.colors.colorWhite
    };
  }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div`
  ${() => css`
  ${Title}{
    position: relative;
    left: 5rem;
  }
  ${Title}::before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -2rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
