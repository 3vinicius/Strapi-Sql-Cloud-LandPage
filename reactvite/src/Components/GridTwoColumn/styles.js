import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Container as Texts } from '../TextComponent/styles';
import darkMode from '../../Utils/stylesUtils.js';

// primaryColor: string;
//         secondaryColor: string;
//         colorWhite: string;

export const Container = styled.div`
  ${({ theme, background }) => css`

    display:grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media(max-width: 850px){
      grid-template-columns: 1fr;
      text-align: center;
    }


    ${Texts}{
      ${darkMode(background)}
    }
    ${Title}{
      margin-bottom: ${theme.spacings.large};
      ${darkMode(background)}
    }

  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  @media ${theme.myMedias.letMedium}{
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImgContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Img = styled.img`
  ${({ theme }) => css`
  width: 100%;
  min-width: 200px;
  overflow: hidden;
  @media(max-width: 850px){
      width: 400px;
    }
  @media ${theme.myMedias.letMedium}{
    width:300px;
  }
  `}
`;
