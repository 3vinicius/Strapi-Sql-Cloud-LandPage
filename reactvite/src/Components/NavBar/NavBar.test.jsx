import { screen } from '@testing-library/react';
import { NavBar } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock.links';
import { theme } from '../../styles/theme';

describe('<NavBar />', () => {
  it('Should render <NavBar />', () => {
    const { container } = renderTheme(<NavBar links={mock} />);
    const child = screen.getByText(mock[1].children);
    expect(container).toBeInTheDocument();
    expect(child).toBeInTheDocument();

    const tagName = container.querySelectorAll('a');
    expect(tagName[1]).toBeVisible();
  });

  it('Should render defautl mode', () => {
    renderTheme(<NavBar />);
    const container = screen.queryByAltText('Main menu');
    expect(container).toBeNull();
  });

  it('Should render correct styles from <NavBar />', () => {
    const { container } = renderTheme(<NavBar links={mock} />);

    const child = container.querySelector('ul');
    expect(child).toHaveStyle({
      'background-color': 'white',
      display: 'flex',
      'flex-direction': 'row',
      'flex-wrap': 'wrap',
    });
  });

  it('Should render correct styles from <NavBar /> media screen', () => {
    renderTheme(<NavBar links={mock} />);
    // const child = container.querySelector('ul');
    expect(screen.getByText(/link2/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      {
        media: theme.myMedias.letMedium,
      },
    );
  });
  it('Should render correct snapshot', () => {
    const { container } = renderTheme(<NavBar links={mock} />);
    expect(container.querySelector('ul')).toMatchSnapshot();
  });
});
