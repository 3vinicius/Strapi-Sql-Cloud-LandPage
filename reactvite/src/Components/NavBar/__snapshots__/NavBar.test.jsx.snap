// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Should render correct snapshot 1`] = `
.c1 {
  display: block;
  color: #0A1128;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  background-color: #DC143C;
  height: 0.2rem;
  width: 0%;
  bottom: 10px;
  left: 50%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover:after {
  left: 25%;
  width: 50%;
}

.c0 {
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<ul
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target2"
    target="_blank"
  >
    link2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_blank"
  >
    link3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_blank"
  >
    link4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_blank"
  >
    link5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_blank"
  >
    link6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_blank"
  >
    link7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_blank"
  >
    link8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_blank"
  >
    link9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_blank"
  >
    link10
  </a>
</ul>
`;
